catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactory.java

```diff
@@ -0,0 +1,340 @@
+                                .getNamespaceContext());
+
+        PropertyName propName = new AttributeExpressionImpl(new NameImpl(name));
+
+        return new SortByImpl(propName, sortBy.getSortOrder());
+    }
+
+    private SortBy[] parseSortBy(SortByType sortByType) throws CswException {
+        JAXBElement<SortByType> sortByElement =
+                new net.opengis.filter.v_1_1_0.ObjectFactory().createSortBy(sortByType);
+
+        return (SortBy[]) parseJaxB(sortByElement);
+    }
+
+    private Object parseJaxB(JAXBElement<?> element) throws CswException {
+        Parser parser = new Parser(PARSER_CONFIG);
+        InputStream inputStream = null;
+
+        try {
+            inputStream = marshalJaxB(element);
+            return parser.parse(inputStream);
+        } catch (JAXBException e) {
+            throw new CswException("Failed to parse Element: (JAXBException): " + e.getMessage(),
+                    CswConstants.INVALID_PARAMETER_VALUE,
+                    null);
+        } catch (IOException e) {
+            throw new CswException("Failed to parse Element: (IOException): " + e.getMessage(),
+                    CswConstants.INVALID_PARAMETER_VALUE,
+                    null);
+        } catch (SAXException e) {
+            throw new CswException("Failed to parse Element: (SAXException): " + e.getMessage(),
+                    CswConstants.INVALID_PARAMETER_VALUE,
+                    null);
+        } catch (ParserConfigurationException e) {
+            throw new CswException(
+                    "Failed to parse Element: (ParserConfigurationException): " + e.getMessage(),
+                    CswConstants.INVALID_PARAMETER_VALUE,
+                    null);
+        } catch (RuntimeException e) {
+            throw new CswException("Failed to parse Element: (RuntimeException): " + e.getMessage(),
+                    CswConstants.INVALID_PARAMETER_VALUE,
+                    null);
+        }
+    }
+
+    private InputStream marshalJaxB(JAXBElement<?> filterElement) throws JAXBException {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        getJaxBContext().createMarshaller()
+                .marshal(filterElement, os);
+        ByteArrayInputStream input = new ByteArrayInputStream(os.toByteArray());
+        IOUtils.closeQuietly(os);
+
+        return input;
+    }
+
+    public static synchronized JAXBContext getJaxBContext() throws JAXBException {
+        if (jaxBContext == null) {
+
+            jaxBContext = JAXBContext.newInstance("net.opengis.cat.csw.v_2_0_2:"
+                            + "net.opengis.filter.v_1_1_0:net.opengis.gml.v_3_1_1:net.opengis.ows.v_1_0_0",
+                    CswQueryFactory.class.getClassLoader());
+        }
+        return jaxBContext;
+    }
+
+    private Filter parseFilter(FilterType filterType) throws CswException {
+        if (!filterType.isSetComparisonOps() && !filterType.isSetId() && !filterType.isSetLogicOps()
```

Comment:
unfortunately no the CswFilterFactory has a private method also does this but I'm not sure thats the right place to do it either since it really doesn't deal with validation